sdk/src/VeraxSdk.ts

```diff
@@ -144,16 +154,25 @@ export class VeraxSdk {
   public utils: UtilsDataMapper;
 
   constructor(conf: Conf, publicAddress?: Address, privateKey?: Hex) {
+    const transport: Transport =
+      conf.mode === SDKMode.FRONTEND
+        ? conf.rpcUrl
+          ? http(conf.rpcUrl)
+          : custom(window.ethereum)
+        : conf.rpcUrl
+        ? http(conf.rpcUrl)
+        : http();
```

Comment:
This is unreadable 😉 
You should try something like 
```suggestion
let transport: Transport;

if (conf.rpcUrl) {
  transport = http(conf.rpcUrl);
} else if (conf.mode === SDKMode.FRONTEND && typeof window.ethereum !== "undefined") {
  transport = custom(window.ethereum);
} else {
  transport = http();
}
```

Reply:
solved it